src/tests/Common/CLRTest.Execute.Bash.targets

```diff
@@ -366,7 +366,56 @@ fi
 $__Command $HARNESS_RUNNER android test --instrumentation="net.dot.MonoRunner" --package-name="net.dot.$__Category" --app="$__TestBinaryBase/$__Category.apk" --output-directory=`pwd` --arg=entrypoint:libname=$(MsBuildProjectName).dll --expected-exit-code=100 -v
 CLRTestExitCode=$?
 
-# Exist code of xharness is zero when tests finished successfully
+# Exit code of xharness is zero when tests finished successfully
+CLRTestExpectedExitCode=0
+      ]]>
+      </BashCLRTestLaunchCmds>
+      <BashCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun' And $(TargetOS) == 'iOSSimulator' ">
+      <![CDATA[
+__Command=""
+if [ ! -z ${__TestDotNetCmd+x} ] %3B then
+  __Command+=" $__TestDotNetCmd"
+else
+  __Command+=" dotnet"
+fi
+
+if [ ! -z "$XHARNESS_CLI_PATH" ]; then
+	# When running in CI, we only have the .NET runtime available
+	# We need to call the XHarness CLI DLL directly via dotnet exec
+	HARNESS_RUNNER="exec $XHARNESS_CLI_PATH"
+else
+	HARNESS_RUNNER="xharness"
+fi
+
+xcode_path="%24(dirname "%24(dirname "%24(xcode-select -p)")")"
+simulator_app="$xcode_path/Contents/Developer/Applications/Simulator.app"
+open -a "$simulator_app"
+helix_runner_uid="%24(id -u)"
+sudo launchctl asuser "$helix_runner_uid" $__Command $HARNESS_RUNNER apple run %5c
```

Comment:
For android we introduced separate install/run/uninstall commands so we don't install/uninstall the app a thousand times during the test run which helped improve the time a lot, we probably need the same here: https://github.com/dotnet/xharness/issues/397

Reply:
We have the same commands available in iOS now: https://github.com/dotnet/xharness/blob/98970876f4d21e6da5198a5c41de51ab1101089f/tests/integration-tests/Apple/Simulator.Commands.Tests.proj#L24-L41

https://github.com/dotnet/xharness/issues/493

Reply:
hmm it's a pity that the `android run` command does run-without-install but the `apple run` command does install+run+uninstall and you need `apple just-test`. can we look at coming up with some unified naming in xharness?

Reply:
Yeah, there's a discrepancy because Apple has two different runtime modes - when it just runs the app and when it does the whole TestRunner listening and whatnot.

I think it would make sense to rename the Android command to `just-test` and have `test` and `just-test` in Android and Apple and then the `run` and `just-run` variants in Apple only?

Reply:
> For android we introduced separate install/run/uninstall commands so we don't install/uninstall the app a thousand times during the test run which helped improve the time a lot, we probably need the same here: [dotnet/xharness#397](https://github.com/dotnet/xharness/issues/397)

I don't think `xharness android install` is being used at all right now by the runtime tests.

Reply:
When https://github.com/dotnet/runtime/pull/54543 gets merged, you should be able to use test only(no more app installation needed here) apple xharness command here.